### PR TITLE
Update listener docs

### PR DESCRIPTION
**Pull request by akshayknarayan**
Following comments in https://github.com/tokio-rs/tokio/pull/2275, clarify the documentation of impl Stream for TcpListener and UnixListener.

### CODE REVIEW

**Comment by akshayknarayan** [comment]
cc @Darksonn @jonhoo

File: tokio/src/net/tcp/listener.rs [hunk 1]
```diff
@@ -12,8 +12,9 @@ use std::task::{Context, Poll};
 cfg_tcp! {
     /// A TCP socket server, listening for connections.
     ///
-    /// Also implements a stream over the connections being received on this listener.
-    ///
+    /// You can accept a new connection by using the accept method. Alternatively this type
+    /// implements the `Stream` trait, which allows you to use the listener in places that want a
+    /// stream.
```
Comment (jonhoo) [inline tokio/src/net/tcp/listener.rs]: This should probably wrap with the text on the following line.

File: tokio/src/net/tcp/listener.rs [hunk 2]
```diff
@@ -278,6 +279,8 @@ impl TcpListener {
 
     /// Returns a stream over the connections being received on this listener.
     ///
+    /// Note that this type also directly implements `Stream`.
```
Comment (jonhoo) [inline tokio/src/net/tcp/listener.rs]: "this type" should probably give the type explicitly.

File: tokio/src/net/unix/listener.rs [hunk 3]
```diff
@@ -15,8 +15,9 @@ use std::task::{Context, Poll};
 cfg_uds! {
     /// A Unix socket which can accept connections from other Unix sockets.
     ///
-    /// Also implements a stream over the connections being received on this listener.
-    ///
+    /// You can accept a new connection by using the accept method. Alternatively this type
+    /// implements the `Stream` trait, which allows you to use the listener in places that want a
+    /// stream.
```
Comment (jonhoo) [inline tokio/src/net/unix/listener.rs]: Wrap

File: tokio/src/net/unix/listener.rs [hunk 4]
```diff
@@ -137,6 +138,8 @@ impl UnixListener {
 
     /// Returns a stream over the connections being received on this listener.
     ///
+    /// Note that this type also directly implements `Stream`.
```
Comment (jonhoo) [inline tokio/src/net/unix/listener.rs]: Should name type

File: tokio/src/net/tcp/listener.rs (hunk 1 above)
Comment (Darksonn) [reply tokio/src/net/tcp/listener.rs]: We might as well also make `accept` and `Stream` links.

**Pull request merged**